pkg/codegen/testing/test/testdata/simple-enum-schema/go/plant/x/provider.go

```diff
@@ -0,0 +1,68 @@
+// Code generated by test DO NOT EDIT.
+// *** WARNING: Do not edit by hand unless you're certain you know what you are doing! ***
+
+package plant
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumix"
+	"simple-enum-schema/plant/internal"
+)
+
+type Provider struct {
+	pulumi.ProviderResourceState
+}
+
+// NewProvider registers a new resource with the given unique name, arguments, and options.
+func NewProvider(ctx *pulumi.Context,
+	name string, args *ProviderArgs, opts ...pulumi.ResourceOption) (*Provider, error) {
+	if args == nil {
+		args = &ProviderArgs{}
+	}
+
+	opts = internal.PkgResourceDefaultOpts(opts)
+	var resource Provider
+	err := ctx.RegisterResource("pulumi:providers:plant", name, args, &resource, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &resource, nil
+}
+
+type providerArgs struct {
+}
+
+// The set of arguments for constructing a Provider resource.
+type ProviderArgs struct {
+}
+
+func (ProviderArgs) ElementType() reflect.Type {
+	return reflect.TypeOf((*providerArgs)(nil)).Elem()
+}
+
+type ProviderOutput struct{ *pulumi.OutputState }
+
+func (ProviderOutput) ElementType() reflect.Type {
+	return reflect.TypeOf((*Provider)(nil)).Elem()
+}
+
+func (o ProviderOutput) ToProviderOutput() ProviderOutput {
+	return o
+}
+
+func (o ProviderOutput) ToProviderOutputWithContext(ctx context.Context) ProviderOutput {
+	return o
+}
```

Comment:
For the generic variant of these output types, we likely don't need these two methods. Is there any reason to keep them?

Reply:
To be honest I am not entirely sure if they are needed. I know the marshalling logic in the core SDK sometimes needs to call `To{TypeName}OutputWithContext(...)` using reflection but whether it is still the case with the generic types, is unclear

Reply:
Good point. Let's follow-up on this separately.